### DISCUSSION TITLE
Federation Invite Code of DNS

### DISCUSSION THREAD

**Discussion by benthecarman** [fedimint/fedimint#4078] [General]
Having long federation bech32m invite codes can be daunting for some users and domain names are much more comfortable for them. Would be neat to have a standard around a simpler way to get the invite code.

Simplest way to do this would be to make a TXT record with the invite code on a given domain. Something more robust might be better however

**Comment by elsirion** [comment]
We could easily add a way to join using one of the guardians' API address, which often boils down to just a domain with `wss://` in front (which you could automatically add). That one guardian could give the user a wrong config though, for a federation where he controls all the nodes. This is why we have invite codes in the first place, they contain data to verify the config fetched from any of the guardians.

The TXT record idea sounds a bit over-complicated given we'd be back at a single point of failure anyway. The main upside would be that it could be hosted by a single party that is more trustworthy than any single of the guardians (but for some reason isn't a guardian).
